src/ParallelAlgorithms/Interpolation/Actions/DumpInterpolatorVolumeData.hpp

```diff
@@ -76,19 +78,22 @@ struct DumpInterpolatorVolumeData {
       db::DataBox<DbTagList>& box,
       const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
       Parallel::GlobalCache<Metavariables>& cache,
-      const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
+      const ArrayIndex& array_index, const ActionList /*meta*/,
       const ParallelComponent* const /*meta*/) {
-    auto& observer_writer = Parallel::get_parallel_component<
-        observers::ObserverWriter<Metavariables>>(cache);
-    auto& my_proxy = Parallel::get_parallel_component<ParallelComponent>(cache);
-    const auto& file_prefix =
-        Parallel::get<observers::Tags::VolumeFileName>(cache);
-    const size_t my_node =
-        Parallel::my_node<size_t>(*Parallel::local_branch(my_proxy));
-    const std::string filename{file_prefix + std::to_string(my_node)};
+    if (not db::get<intrp::Tags::DumpVolumeDataOnFailure>(box)) {
```

Comment:
Add `const_global_cache_tags = tmpl::list<intrp::Tags::DumpVolumeDataOnFailure>` to this action. Then you don't have to worry about this elsewhere.